include/deal.II/lac/trilinos_vector.h

```diff
@@ -500,8 +500,12 @@ namespace TrilinosWrappers
       /**
        * Move constructor. Creates a new vector by stealing the internal data
        * of the vector @p v.
+       *
+       * @note In order for this constructor to leave the moved-from object in a
+       * valid state it must allocate memory (in this case, an empty
+       * Epetra_FEVector) - hence it cannot be marked as noexcept.
```

Comment:
I'm not sure what the exact properties of `Epetra_FEVector`s are but our concern is one level higher - the default constructor for our vector class contains
```cpp
vector(new Epetra_FEVector(
  Epetra_Map(0, 0, 0, Utilities::Trilinos::comm_self())))
```
i.e., an invariant maintained by this class is that `vector` is not nullptr. Hence, even creating an empty object allocates memory.

It's still convenient to have the move ctor since we can swap contents with an allocated (but empty) vector.

Reply:
I see about the invariant.

If the `noexcept` part were important, you can of course always move the pointer from the rhs object, and then use `try{...} catch(...){abort();}` to allocate an empty vector for the rhs. But I don't care either way -- we can deal with this whenever someone actually needs these move operators to be `noexcept`.